analysis/wwz/wwz4l.py

```diff
@@ -452,9 +477,31 @@ def process(self, events):
 
             # Clean with dr (though another option is to use jetIdx)
             cleanedJets = os_ec.get_cleaned_collection(l_wwz_t,jets)
+            jetptname = "pt_nom" if hasattr(cleanedJets, "pt_nom") else "pt"
+
+            ##### JME Stuff #####
+
+            cleanedJets["pt_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.pt
+            cleanedJets["mass_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.mass
+
+            if not isData:
+                cleanedJets["pt_gen"] =ak.values_astype(ak.fill_none(cleanedJets.matched_gen.pt, 0), np.float32)
+            else:
+                cleanedJets["pt_gen"] =ak.ones_like(cleanedJets.pt)
+
+            if year in ["2022","2022EE","2023","2023BPix"]:
+                cleanedJets["rho"] = ak.broadcast_arrays(events.Rho.fixedGridRhoFastjetAll, cleanedJets.pt)[0]
+            elif year in ["2016APV","2016","2017","2018"]:
+                cleanedJets["rho"] = ak.broadcast_arrays(events.fixedGridRhoFastjetAll, cleanedJets.pt)[0]
+
```

Comment:
Can you add a comment to the code about this events_cache thing? I.e. just like one line saying something about what it is or what it does

Reply:
events_cache is used to store intermediate corrections calculations (as the JEC corrections are a step-by-step process)